### PR TITLE
Remove dead code

### PR DESCRIPTION
**Pull request by danmoseley**
1. Remove dead creds from S.DS.AM. These were creds for some long gone local account on a long gone machine. This reduces credential scan noise.
2. Outerloop process tests create a temporary local account, when running elevated, then delete it. Change to a secure password. This also reduces credential scan noise.
3. Remove `-p` going to mkdir, apparently copied from Unix. Windows doesn't have this flag and it doesn't matter for the test.

### CODE REVIEW

File: src/libraries/System.DirectoryServices.AccountManagement/tests/PrincipalTest.cs [hunk 1]
```diff
@@ -15,9 +15,8 @@ public abstract class PrincipalTest : IDisposable
 
         private void RefreshContext()
         {
-            // [SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine", Justification="Unit test password.")]
-            string username = "Administrator", password = "Adrumble@6";
-
+            string username = "Administrator";
+            string password = Environment.GetEnvironmentVariable("TESTPASSWORD");
```
Comment (danmoseley) [inline src/libraries/System.DirectoryServices.AccountManagement/tests/PrincipalTest.cs:19]: These tests don't have a `[Fact]` attribute so they're not even trying to run. But I haven't yet examined whether we should try to run them, so I'm not deleting them. When we do, I believe the best practice for simple creds is to use an environment variable.

File: src/libraries/System.DirectoryServices.AccountManagement/tests/UserPrincipalTest.cs [hunk 2]
```diff
@@ -28,33 +28,5 @@ public void UserPrincipalConstructorTest()
             UserPrincipal user = new UserPrincipal(DomainContext);
             user.Dispose();
         }
-
-        public void ComputedUACCheck()
```
Comment (danmoseley) [inline src/libraries/System.DirectoryServices.AccountManagement/tests/UserPrincipalTest.cs:32]: dead code

Comment (danmoseley) [reply src/libraries/System.DirectoryServices.AccountManagement/tests/UserPrincipalTest.cs:32]: Let me correct that: it's a test of AuthenticablePrincipal.IsAccountLocked out, but it never got a `[Fact]` on it. If we bring a test back, we can reuse code then.

Comment (danmoseley) [reply src/libraries/System.DirectoryServices.AccountManagement/tests/UserPrincipalTest.cs:32]: I'm deleting this one and not the other one as this seems like something we'd always keep as a manual test.

File: src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs [hunk 3]
```diff
@@ -456,8 +457,16 @@ public void TestWorkingDirectoryPropertyInChildProcess()
...
+            const string username = "testForDotnetRuntime";
+            string password;
+            using (RandomNumberGenerator rng = new RNGCryptoServiceProvider())
+            {
+                byte[] randomBytes = new byte[33];
+                rng.GetBytes(randomBytes);
+
+                // Add special chars to ensure it satisfies password requirements.
+               password = Convert.ToBase64String(randomBytes) + "_-As@!%*(1)4#2";
+           }
```
Comment (stephentoub) [inline src/libraries/System.Diagnostics.Process/tests/ProcessStartInfoTests.cs:469]:
```suggestion
            string password = Convert.ToBase64String(RandomNumberGenerator.GetBytes(33)) + "_-As@!%*(1)4#2";
```

**Review by stephentoub — APPROVED** [review]

**Pull request merged**